### PR TITLE
feat(Menu): Prototype

### PR DESCRIPTION
**Pull request by assuncaocharles**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

As agreed this PR adds a **private** prototype for `Menu` with submenus

gLOC5mb5gx

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/fluentui/docs/src/prototypes/menuList/MenuButton.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          style={{
+            border: '1px solid black',
+            width: 55,
+            textAlign: 'center',
+          }}
+        >
+          Menu v
+        </div>
+      </MenuTrigger>
+      <MenuList id="1">
```
Comment (layershifter) [inline packages/fluentui/docs/src/prototypes/menuList/MenuButton.tsx:22]: Do we need `id` now?

File: packages/fluentui/docs/src/prototypes/menuList/Menu.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import { MenuContextProvider } from './menuContext';
+import { useEventListener } from '../../../../react-component-event-listener';
```
Comment (layershifter) [inline packages/fluentui/docs/src/prototypes/menuList/Menu.tsx:3]: Let's fix these imports

File: packages/fluentui/docs/src/prototypes/menuList/MenuButton.tsx (hunk 1 above)
Comment (assuncaocharles) [reply packages/fluentui/docs/src/prototypes/menuList/MenuButton.tsx:22]: Removed

File: packages/fluentui/docs/src/prototypes/menuList/Menu.tsx (hunk 2 above)
Comment (assuncaocharles) [reply packages/fluentui/docs/src/prototypes/menuList/Menu.tsx]: updated

**Comment by assuncaocharles** [comment]
Anything else @layershifter ?

**Comment by assuncaocharles** [comment]
@miroslavstastny , @mnajdova , @pompomon  Maybe you can also check if you have any thoughts on this?

**Review by layershifter — APPROVED** [review]

**Pull request merged**